teps/0048-task-results-without-results.md

```diff
@@ -0,0 +1,293 @@
+    # the image
+    - name: build-image
+      runAfter: [ "check-pr-content" ]
+      when:
+        - input: "$(tasks.check-pr-content.results.image-change)"
+          operator: in
+          values: ["yes"]
+      taskRef:
+        name: build-image
+      results:
+        - name: image-name
+    # deploy a newly built image if build-image was successful and produced an image name
+    # deploy a latest platform by default if there are no changes in this PR
+    - name: deploy-image
+      runAfter: [ "build-image" ]
+      params:
+        - name: image-name
+          value: "$(tasks.build-image.results.image-name.path)"
+      taskRef:
+        name: deploy-image
+    # update the page where a list of builds is maintained with this new image
+    - name: update-list-of-builds
+      runAfter: [ "build-image" ]
+      params:
+        - name: image-name
+          value: "$(tasks.build-image.results.image-name.path)"
+      when:
+        - input: "$(tasks.build-image.status)"
+          operator: in
+          values: ["succeeded"]
+      taskRef:
+        name: update-list-of-builds
+```
+
+### `Pipeline Results` from the conditional tasks
+
+Produce the name of the image as the pipeline result depending on how the image was built.
+
+```yaml
+spec:
+  tasks:
+    # Clone application repo
+    - name: git-clone
+      taskRef:
+        name: git-clone
+    # TRUST_BUILDER is set to true at the pipelineRun level if the builder image is trusted
+    # if the builder image is trusted, executed build-trusted and produce an image name as a result
+    - name: build-trusted
+      runAfter: [ "git-clone" ]
+      when:
+        - input: "$(params.TRUST_BUILDER)"
+          operator: in
+          values: ["true"]
+      taskRef:
+        name: build-trusted
+      results:
+        - name: image
+    # TRUST_BUILDER is set to false at the pipelineRun level if the builder image is not trusted
+    # and needs to run in isolation
+    # if the builder image is not trusted, executed build-un trusted and produce an image name as a result
+    - name: build-untrusted
+      runAfter: [ "git-clone" ]
+      when:
+        - input: "$(params.trusted)"
+          operator: in
+          values: ["false"]
+      taskRef:
+        name: build-untrusted
+      results:
+        - name: image
+    # read result of both build-trusted and build-untrusted and propagate the one which is initialized as a pipeline result
+    - name: propagate-image-name
+      runAfter: [ "build-image" ]
+      params:
+        - name: trusted-image-name
+          value: "$(tasks.build-trusted.results.image)"
+        - name: untrusted-image-name
+          value: "$(tasks.build-untrusted.results.image)"
+      taskRef:
+        name: propagate-image-name
+      results:
+        - name: image
+  # pipeline result
+  results:
+    - name: APP_IMAGE
+      value: $(tasks.propagate-image-name.results.image)
+```
+
```

Comment:
Is it correct to think, another possible use case can be, for resuming failed pipelines.

The tasks that ran fine can be somehow disabled (either with some future TEP or when expression), and their results pre-populated, while copying and creating a new PipelineRun in order to retry/resume.

Reply:
that's an interesting use case for this @ScrapCodes 🤔 

my feeling is that in the resuming a pipeline case, usually youd want to use the result value produced by the previous run, and not the default value - tho maybe sometimes you would?

for example, say you were trying to do a build + deploy:

```yaml
spec:
  tasks:
    - name: build-image
      taskRef:
        name: build-image
      results:
        - name: image-name
    - name: deploy-image
      params:
        - name: image-name
          value: "$(tasks.build-image.results.image-name)" # using the image that was built by build-image
      taskRef:
        name: deploy-image
```

Maybe `deploy-image` failed, and you want to run the pipeline again, but you don't want to have to build again, you probably want to use the value of ` tasks.build-image.results.image-name` from the previous run.

If you used the functionality proposed here, you'd be using some default value for the image-name instead of the value that was just built.  For example, imagining a syntax for specifying a default to use:

```yaml
spec:
  tasks:
    - name: build-image
      taskRef:
        name: build-image
      results:
        - name: image-name
    - name: deploy-image
      params:
        - name: image-name
          value: "$(tasks.build-image.results.image-name?my-website:v0.4.0)"
      taskRef:
        name: deploy-image
```

Let's say this built an image `my-website:v0.7.0`, but `deploy-image` failed. If you wanted to re-run, i.e. partially run the pipeline with `build-image` disabled, I'm guessing you'd want to pass `my-website:v0.7.0` (the one you just built) and not use the default `my-website:v0.4.0`

Probably there are cases for both tho!